### PR TITLE
rgw ldap: fix ldap bindpw parsing

### PR DESCRIPTION
**Pull request by mattbenjamin**
Fixes DN search issue using QE configuration of MS AD.

Signed-off-by: Matt Benjamin <mbenjamin@redhat.com>

### CODE REVIEW

File: src/rgw/rgw_rest_s3.cc [hunk 1]
```diff
@@ -2922,9 +2946,28 @@ void RGW_Auth_S3::init_impl(RGWRados* store)
...
+    int pwlen = safe_read_file("" /* base */,
+			       store->ctx()->_conf->rgw_ldap_secret.c_str(),
+			       bindpw, 1023);
+    if (pwlen) {
+      ldap_bindpw = bindpw;
+      boost::algorithm::trim(ldap_bindpw);
+      if (ldap_bindpw.back() == '\n')
+	ldap_bindpw.pop_back();
+    }
+  }
```
Comment (cbodley) [inline src/rgw/rgw_rest_s3.cc]: looks like this block is duplicated in librgw.cc, consider moving into a helper function in rgw_ldap.h

**Comment by mattbenjamin** [comment]
ok, helper function added

**Comment by mattbenjamin** [comment]
helper function added

**Comment by smithfarm** [comment]
Changelog:
* opened tracker issue to enable backporting workflow
* added tracker issue link to PR description

**Pull request merged**